core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -194,4 +256,42 @@ public ACL getACL() {
     public static final Permission PROVISION = new Permission(
             Computer.PERMISSIONS, "Provision", Messages._Cloud_ProvisionPermission_Description(), Jenkins.ADMINISTER, PERMISSION_SCOPE
     );
+
+    /**
+     * Parameter object for {@link hudson.slaves.Cloud}.
+     * @since XXX
+     */
+    public static final class CloudState {
+        /**
+         * The label under consideration.
+         */
+        @CheckForNull
+        private final Label label;
+        /**
+         * The additional planned capacity for this {@link #label} and provisioned by previous strategies during the
+         * current updating of the {@link NodeProvisioner}.
+         */
+        private final int additionalPlannedCapacity;
+
+        public CloudState(@CheckForNull Label label, int additionalPlannedCapacity) {
+            this.label = label;
+            this.additionalPlannedCapacity = additionalPlannedCapacity;
+        }
+
+        /**
+         * The label under consideration.
+         */
+        @CheckForNull
+        public Label getLabel() {
+            return label;
+        }
+
+        /**
+         * The additional planned capacity for this {@link #getLabel()} and provisioned by previous strategies during
+         * the current updating of the {@link NodeProvisioner}.
+         */
+        public synchronized int getAdditionalPlannedCapacity() {
```

Comment:
Curious as to why this has to be synchronized

Reply:
It doesn't need to be, this comes from copy-paste from `StrategyState`. In `CloudState` it's just readonly so I'll remove it.